.github/workflows/release.yml

```diff
@@ -66,3 +66,34 @@ jobs:
           artifactContentType: application/zip
           draft: true
           generateReleaseNotes: true
+
+  create-npm-package:
+    runs-on: ubuntu-latest
+    if: ${{ github.repository == 'quisquous/cactbot' }}
+    steps:
+      - name: Checkout Code
+        uses: actions/checkout@v3
+      - uses: ./.github/actions/setup-js-env
+
+      - run: mkdir npm-package
+
+      - name: Build JavaScript UI Module Bundles
+        run: npm run build
+
+      - run: mv dist npm-package/
+
+      - run: npx tsc --declaration
+
+      - run: |
+          cp -r types/./ npm-package/types/
+          cp -r ./dist/ui/./ npm-package/ui/
+          cp -r ./dist/resources/./ npm-package/resources/
```

Comment:
My understanding here is that you don't want to publish util, test, and webpack, which seems good.

Are you also doing this because you want `npm-package/dist/ui/` to be the transpiled javascript files and `npm-package/ui/` to be the extra types (that aren't in `types`)? Is that normally how a typescript package is supposed to be set up?

(Also, for my own knowledge, what does `/./` do here? It is hard to search for this to understand what it does.)

(Also also, for consistency should `cp -r types/./ etc` be `cp -r ./types/./ etc`? (Not sure why some of these are `./` relative but not types.)

Reply:
No, here `npm-package/dist/` is result of `npm run build`

and `npm-package/ui/` is transpiled javascript files.

And some of transpiled `.d.ts` need to import types from `npm-package/types/`